### ISSUE TITLE
Failure to re-render old tutorials

### ISSUE THREAD

**Issue by bborgesr** [rstudio/learnr#151]
**TLDR**: 
- **_the problem_: tutorial fails to render with obscure error**
- **_the quick fix_: delete old .html file and re-run tutorial**

## The problem

In a nutshell, the problem is trying to re-render an unchanged old tutorial with updated versions of `rmarkdown` and `learnr`, while still preserving the old .html artifact from a previous run. Here's step-by-step workflow that is unfortunately likely to result in an obscure error. If you:

1. have a `learnr` tutorial (or any `runtime: shiny_prerendered` .Rmd document) **AND**
2. had previously run that .Rmd document **AND** 
3. preserved the .html artifact **AND**
4. recently updated the `learnr` and `rmarkdown` packages **AND**
5. tried to re-run the .Rmd without any changes **AND**
6. encountered an obscure error involving failing to normalize a path...

... then, this issue is for you. This is fairly common and even encouraged workflow for tutorial authors, so it's regrettable that such a common workflow could cause such an obscure error. We've improved the error message in the development version of Shiny (version > 1.0.5), but this is still far from ideal.

Here's what you may see for Shiny version 1.0.5 (or lower), both in the R Markdown console, and in the viewer/window (i.e. where the tutorial or app usually appears):

```sh
Warning: Error in normalizePath: path[1]="": No such file or directory    # in Rmd console
ERROR: path[1]="": No such file or directory                              # in app viewer
```

![image](https://user-images.githubusercontent.com/6527540/37715966-49db59d2-2d15-11e8-8217-f26f88a1a645.png)

And here's what you may see for Shiny version greater than 1.0.5 (currently only available from github: `devtools::install_github("rstudio/shiny")`) -- in this case, both the R Markdown console and the viewer give the same, slightly more descriptive error:

```sh
ERROR: Couldn't normalize path in `addResourcePath`, with arguments: 
  `prefix` = 'highlightjs-1.1'; `directoryPath` = ''
```

![image](https://user-images.githubusercontent.com/6527540/37715933-2f6b1dbc-2d15-11e8-9cbf-5844f9a476ab.png)

## The quick fix
The very high-level reason for this bug is that the old .html file cached some now out-of-date dependencies, which the current version of `rmarkdown` tries to load, but fails to. For more details on this, see the next section. For this section, we only care about the fact that this problem is caused by the old .html file generated by a previous run. This means that:

- deleting the .html file **OR**
- moving the .html file to another directory **OR** 
- even just changing the .Rmd document (so as to force it to knit again) 

**AND** then running it bypasses this problem at all. Either of these things is a way to force the .Rmd document to be knit from scratch again, and so it no longer tries to use the cached .html file.

The glass half-empty side of this bug is that it exists at all and that you, the app author, have to deal with it. The glass half-full side is that dealing with it is very easy, once you know it: just delete the old .html file.

## The hint of a better explanation
This is an issue that can actually affect any `runtime: shiny_prerendered` .Rmd document. The root source of this problem is in the `rmarkdown` package and it is documented at https://github.com/rstudio/rmarkdown/issues/1278. Once that issue is addressed, this type of problem should disappear completely in future releases. And I'll close this issue here too.

The particular problem you saw above (involving 'highlightjs-1.1') was monkey patched in rmarkdown 1.8.9 (rmakrdown 1.9 is now avaibale on CRAN): https://github.com/rstudio/rmarkdown/commit/d3331691947add59b20c65459e1f3a0b3f15173e. However, this only "takes off" after the .html file is regenerated. The easiest way to do this is simply to delete the old .html file and re-run the .Rmd tutorial from scratch.

There's a chance there will be other dependency changes causing similar errors before this issue is fully addressed on the `rmarkdown` side. Regenerating the .html file from scratch should resolve the issue. If not, please comment below or open a new issue.

**Comment by JunlueZhao** [comment]
Hi Barbara, since this issue is still opened, does that mean that this issue hasn't been fixed yet?

**Comment by j3ypi** [comment]
I shipped my tutorial within a package. Does this mean, I have to delete the .html file every time `rmarkdown` or `learnr` recieves an update?
